src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp

```diff
@@ -845,39 +845,31 @@ inline void exp_reduce_sum(float* a, const float max, const size_t size, float&
 inline void exp_reduce_sum_f32(ov::float16* a, const ov::float16 max, const size_t size, ov::float16& sum) {
     size_t i = 0;
 #    if defined(HAVE_SVE)
-    svfloat32_t v_a;
     svfloat32_t v_max = svdup_n_f32(static_cast<float>(max));
     svfloat32_t v_sum = svdup_n_f32(0.0f);
 
     svbool_t pg_f32 = svptrue_b32();
     svbool_t pg_f16 = svptrue_b16();
     svfloat16_t zero = svdup_n_f16(0.0);
-    size_t inc = vec_len_f32_sve();
 
-    while (i < size) {
-        if (size - i < vec_len_f16_sve())
-            pg_f16 = svwhilelt_b16(0, static_cast<int>(size - i));
-        if (size - i < vec_len_f32_sve()) {
-            pg_f32 = svwhilelt_b32(0, static_cast<int>(size - i));
-            inc = size - i;
-        }
-        // Load 16 elements and interleave with zeros so we have 8 elements with 0 in high parts
+    for (; i + svcnth() < size; i += svcnth()) {
```

Comment:
It seems to me, that `i + svcnth() < size` skips the last full block (or the only block if `size == svcnth()`), so this block runs through scalar path. Shouldn't we change the condition to `i + svcnth() <= size`?